File: 1901/demo/shell_bronze.java

import java.io.*;
import java.util.*;

public class shell_bronze {

    public static void main(String[] args) throws IOException
    {
        int N;
        Scanner In = new Scanner(new File("shell.in"));
        N = In.nextInt();
        int[] A = new int[N];
        int[] B = new int[N];
        int[] G = new int[N];
        for (int  i = 0; i<N; i++) {
            A[i] = In.nextInt();
            B[i] = In.nextInt();
            G[i] = In.nextInt();
            A[i]--;
            B[i]--;
            G[i]--;
        }
        int[] poss = new int[3];
        int[] wins = new int[3];
        for (int  j = 0; j<3; j++) {
            poss[j] = j;
        }
        for (int  i = 0; i<N; i++) {
            for (int  j = 0; j<3; j++) {
                if (poss[j] == A[i]) {
                    poss[j] = B[i];
                } else if (poss[j] == B[i]) {
                    poss[j] = A[i];
                }
                if (G[i] == poss[j]) {
                    wins[j]++;
                }
            }
        }
        int max = 0;
        for (int  j = 0; j<3; j++) {
            if (max < wins[j]) {
                max = wins[j];
            }
        }
        PrintWriter Out = new PrintWriter(new File("shell.out"));
        Out.printf("%d\n", max);
        Out.close();
    }
}

File: 1901/demo/shell_bronze.cpp
#include <cstdio>
#include <cstdio>
#include <algorithm>

int main(int argc, char** argv)
{
    FILE* In = fopen("shell.in", "r");
    if (!In) {
        fprintf(stderr, "Can't open In\n");
        exit(-1);
    }
    int N;
    fscanf(In, " %d" ,&N);
    int A[N];
    int B[N];
    int G[N];
    for (int  i = 0; i<N; i++) {
        fscanf(In, " %d %d %d" ,&A[i],&B[i],&G[i]);
        A[i]--;
        B[i]--;
        G[i]--;
    }
    fclose(In);
    int poss[3] = {0, 1, 2};
    int wins[3] = {0};
    for (int  i = 0; i<N; i++) {
        for (int  j = 0; j<3; j++) {
            if (poss[j] == A[i]) {
                poss[j] = B[i];
            } else if (poss[j] == B[i]) {
                poss[j] = A[i];
            }
            if (G[i] == poss[j]) {
                wins[j]++;
            }
        }
    }
    int max = 0;
    for (int  j = 0; j<3; j++) {
        if (max < wins[j]) {
            max = wins[j];
        }
    }
    FILE* Out = fopen("shell.out", "w");
    fprintf(Out, "%d\n", max);
    fclose(Out);
    return 0;
}
